src/Framework/Framework/Resources/Scripts/state-manager.ts

```diff
@@ -327,7 +327,7 @@ function createWrappedObservable<T>(initialValue: DeepReadonly<T>, typeHint: Typ
                     if (newContents[index] && newContents[index][notifySymbol as any]) {
                         continue
                     }
-                    if (compileConstants.debug && newContents[index]) {
+                    if (compileConstants.debug && ko.isObservable(newContents[index])) {
```

Comment:
I think we should warn when the object contains any observables. The problem is that DotVVM replaces all of them, so if your observables won't get updated.